cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-expandable.html

```diff
@@ -43,16 +43,10 @@
             <span class="financial-item_value" {{ value.data_attribute | safe if value.data_attribute else '' }} >
                {{ value.value }}
             </span>
-            <span class="o-expandable_cue o-expandable_cue-open">
-                <span class="u-visually-hidden-on-mobile">
-                    {{ 'Edit' if value.is_editable else 'Show' }}
-                </span>
+            <span class="o-expandable_cue-open" aria-label="{{ 'Edit' if value.is_editable else 'Show' }}">
```

Comment:
Did you mean to make these translatable?

```suggestion
            <span class="o-expandable_cue-open" aria-label="{{ _('Edit') if value.is_editable else _('Show') }}">
```

I'm a little wary of whether using an `if` statement might violate [this guidance](https://cfpb.github.io/consumerfinance.gov/translation/#troubleshooting), though.

Reply:
Yeah I left them at the status quo because I wasn't sure how it would work with the `if` statement. Happy to merge your change if you're good with it?

Reply:
No, unfortunately my suggestion doesn't work. If I apply my suggestions and run `makemessages` as documented [here](https://cfpb.github.io/consumerfinance.gov/translation/#2-run-the-makemessages-management-command-to-add-the-string-to-the-portable-object-file), it seems like gettext only finds the first words on the line (`_('Edit')` and `_('Save')`) but misses the second ones (`_('Show')` and `_('Hide')`).

This works, although it's annoyingly verbose:

```diff
diff --git a/cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-expandable.html b/cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-expandable.html
index bf61f31ea..feb646d57 100644
--- a/cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-expandable.html
+++ b/cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-expandable.html
@@ -43,10 +43,17 @@
             <span class="financial-item_value" {{ value.data_attribute | safe if value.data_attribute else '' }} >
                {{ value.value }}
             </span>
-            <span class="o-expandable_cue-open" aria-label="{{ 'Edit' if value.is_editable else 'Show' }}">
+            {%- if value.is_editable -%}
+                {% set aria_label_open = _('Edit') %}
+                {% set aria_label_close = _('Save') %}
+            {%- else -%}
+                {% set aria_label_open = _('Show') %}
+                {% set aria_label_close = _('Hide') %}
+            {%- endif %}
+            <span class="o-expandable_cue-open" aria-label="{{ aria_label_open }}">
                 {{ svg_icon('edit') if value.is_editable else svg_icon('plus-round') }}
             </span>
-            <span class="o-expandable_cue-close" aria-label="{{ 'Save' if value.is_editable else 'Hide' }}">
+            <span class="o-expandable_cue-close" aria-label="{{ aria_label_close }}">
                 {{ svg_icon('approved-round') if value.is_editable else svg_icon('minus-round') }}
             </span>
         </span>
```

Reply:
Ah ok. I'll leave it as-is without the translated bits for now.

Reply:
Ah, I wasn't necessarily saying it was __too__ annoyingly verbose :) I think it's good to be consistent about translating everywhere, if the messy Jinja isn't too bothersome to add.

Reply:
I agree consistency is good, but something consider for these: I can't check the crawler at the moment per the network being down, but I believe these are only used in https://www.consumerfinance.gov/paying-for-college/your-financial-path-to-graduation/. It seems like overall that app would need a lot of translation work attention, well beyond the show/hide, edit/save buttons. If I'm understanding the flow, it would be several screens of english before a user encountered these labels. Additionally, this is a customized expandable that should go away and be its own thing, and a cursory look through the tool I'm only seeing the "Edit" version of these. I think before we add verbosity to these templates, we should evaluate their usage and refactor them.

Reply:
Yeah, it's documented a little bit [under troubleshooting](https://cfpb.github.io/consumerfinance.gov/translation/#troubleshooting) as to why a less verbose but more logically complex expression doesn't work with `makemessages`.

The gist is, gettext doesn't understand Jinja/Django templates, so `makemessages` converts the templates into a quasi-Python format. This... leaves something to be desired, but Jinja allows a *lot* more logic in templates than Django's templates do, and complex Jinja expressions get muddled with that conversion, meaning that gettext can't find the string marked for translation.